### PR TITLE
ToC Organisation

### PR DESCRIPTION
**Pull request by maiyahershey**
This PR syncs the ToC with the .md source files, minor title cleanup, and addresses comments from Kirk's doc.

### CODE REVIEW

File: docs/introduction/infrastructure.md
Comment (sergiomsantos) [inline docs/introduction/infrastructure.md:1]: where did the content of this file go?

File: docs/cli.md
Comment (sergiomsantos) [inline docs/cli.md:1]: the content of this file has been split across different section correct?

Comment (maiyahershey) [reply docs/cli.md:1]: yes, it is now cli-overview in the cli folder

File: docs/introduction/infrastructure.md
Comment (maiyahershey) [reply docs/introduction/infrastructure.md:1]: it now lives in explore api folder as computational-infrastructure.md

**Pull request merged**